### PR TITLE
Update Bloc Library Sample

### PR DESCRIPTION
**Pull request by felangel**
- Improvements to Bloc Library Sample
- Bloc Library Tests Added
- flutter_bloc version bump

### CODE REVIEW

File: bloc_library/pubspec.yaml [hunk 1]
```diff
@@ -7,13 +7,21 @@ environment:
 dependencies:
   meta: ">=1.1.0 <2.0.0"
   equatable: ^0.2.0
-  flutter_bloc: ^0.7.0
+  flutter_bloc: ^0.10.0
   flutter:
     sdk: flutter
+
+dependency_overrides:
```
Comment (felangel) [inline bloc_library/pubspec.yaml]: flutter_bloc ^0.10.0 depends on rxdart ^0.21.0 so this change was made to allow the bloc_library to work without forcing an upgrade in other examples to user rxdart ^0.21.0.

**Comment by felangel** [comment]
Any update on this? Thanks!

**Comment by brianegan** [comment]
Oh my goodness, sorry @felangel! Thought I'd merged this one -- looks great :)

**Comment by felangel** [comment]
No worries haha. Thanks @brianegan! 🙏

**Pull request merged**